subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/DefaultConnection.java

```diff
@@ -63,8 +56,8 @@
 
 import java.io.File;
 
-public class DefaultConnection implements ConnectionVersion4, InternalConnection, BuildActionRunner,
```

Comment:
This class shouldn't change, otherwise you're breaking compatibility with older Gradle versions.